lib/auth/auth_with_roles.go

```diff
@@ -1286,59 +1265,13 @@ func (r resourceChecker) CanAccess(resource types.Resource) error {
 		return r.CheckAccess(rr, mfaParams)
 	case types.WindowsDesktop:
 		return r.CheckAccess(rr, mfaParams)
+	case types.Server:
+		return r.CheckAccess(rr, mfaParams)
 	default:
 		return trace.BadParameter("could not check access to resource type %T", r)
 	}
 }
 
-// nodeChecker is a resourceAccessChecker that checks for access to nodes
-type nodeChecker struct {
-	accessChecker services.AccessChecker
-	builtinRole   bool
-}
-
-// newNodeChecker returns a new nodeChecker that checks access to nodes with the
-// the provided user if necessary. This prevents the need to load the role set each time
-// a node is checked.
-func newNodeChecker(authContext Context, authServer *Server) (*nodeChecker, error) {
-	// For certain built-in roles, continue to allow full access and return
-	// the full set of nodes to not break existing clusters during migration.
-	//
-	// In addition, allow proxy (and remote proxy) to access all nodes for its
-	// smart resolution address resolution. Once the smart resolution logic is
-	// moved to the auth server, this logic can be removed.
-	builtinRole := HasBuiltinRole(authContext, string(types.RoleAdmin)) ||
-		HasBuiltinRole(authContext, string(types.RoleProxy)) ||
-		HasRemoteBuiltinRole(authContext, string(types.RoleRemoteProxy))
```

Comment:
Looks like here we had a special case that allowed these builtin roles to see all the nodes, I assume for resource-based access requests. I don't see this logic anymore, do we need to implement it in `checkAccessToNode`?

cc @nklaassen

Reply:
Not sure if this is being used by resource based access requests.
This particular check comes from this PR https://github.com/gravitational/teleport/pull/2175

I ended up removing this part because
- I don't see another mention of "smart resolution logic", this could mean we should remove this
- tests didn't fail after this removal

Looking back, it might have been an oversight and we might want to add this back (and even add some testing)